public/app/plugins/datasource/prometheus/configuration/PromSettings.tsx

```diff
@@ -130,3 +138,15 @@ const onChangeHandler = (key: keyof PromOptions, value: Props['value'], onChange
     },
   });
 };
+
+const onChangeHandlerSwitch = (key: keyof PromOptions, value: Props['value'], onChange: Props['onChange']) => (
+  eventItem: SyntheticEvent<HTMLInputElement>
+) => {
+  onChange({
+    ...value,
+    jsonData: {
+      ...value.jsonData,
+      [key]: eventItem.currentTarget.checked,
+    },
+  });
+};
```

Comment:
We already have a function for this in grafana/data , onUpdateDatasourceJsonDataOptionChecked